lib/cc/engine/analyzers/file_thread_pool.rb

```diff
@@ -14,11 +14,13 @@ def initialize(files, concurrency: DEFAULT_CONCURRENCY)
 
         def run(&block)
           queue = build_queue
+          lock = Mutex.new
 
           @workers = thread_count.times.map do
             Thread.new do
-              while !queue.empty? && (item = queue.pop(true))
-                yield item
+              until queue.empty?
+                item = lock.synchronize { queue.pop(true) unless queue.empty? }
+                yield item if item
```

Comment:
> I think the lock would need to be a member var, yeah? Getting a new lock on every method call doesn't seem real helpful.

Can you expand on that for my own education? I'm not familiar with this and assumed it didn't matter from where you instantiated the Mutex, it would always work the same.